### PR TITLE
NormaliseByDetectorTestPerformance heap-buffer-overflow fixed

### LINKED ISSUE #17443: Algorithms::NormaliseByDetector: AddressSanitizer: heap-buffer-overflow
From a run of units tests with Address Sanitizer (compiled with `-DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS=-fsanitize=address`):

```
1069/1683 Test #1069: AlgorithmsTest_NormaliseByDetectorTestPerformance ............***Failed    2.98 sec
Running 2 testsFrameworkManager-[Notice] Welcome to Mantid 3.7.20160907.1229
FrameworkManager-[Notice] Please cite: http://dx.doi.org/10.1016/j.nima.2014.07.029 and this release: http://dx.doi.org/10.5286/Software/Mantid
Load-[Notice] Load started
Load-[Notice] Load successful, Duration 1.03 seconds
ConvertUnits-[Notice] ConvertUnits started
ConvertUnits-[Notice] ConvertUnits started
ConvertUnits-[Notice] ConvertUnits successful, Duration 0.02 seconds
ConvertUnits-[Notice] ConvertUnits started
ConvertUnits-[Notice] ConvertUnits successful, Duration 0.02 seconds
ConvertUnits-[Notice] ConvertUnits successful, Duration 0.04 seconds. Processed as a workspace group
ConvertSpectrumAxis-[Notice] ConvertSpectrumAxis started
ConvertSpectrumAxis-[Notice] ConvertSpectrumAxis started
ConvertSpectrumAxis-[Warning] The instrument definition is incomplete - spectra dropped from output
ConvertSpectrumAxis-[Notice] ConvertSpectrumAxis successful, Duration 0.01 seconds
ConvertSpectrumAxis-[Notice] ConvertSpectrumAxis started
ConvertSpectrumAxis-[Warning] The instrument definition is incomplete - spectra dropped from output
ConvertSpectrumAxis-[Notice] ConvertSpectrumAxis successful, Duration 0.01 seconds
ConvertSpectrumAxis-[Notice] ConvertSpectrumAxis successful, Duration 0.02 seconds. Processed as a workspace group
LoadParameterFile-[Notice] LoadParameterFile started
=================================================================
==133619==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x60e00011d5d8 at pc 0x7f24e2f08a65 bp 0x7ffe20be82a0 sp 0x7ffe20be8290
READ of size 8 at 0x60e00011d5d8 thread T0
    #0 0x7f24e2f08a64 in Mantid::Kernel::TimeValueUnit<double>::value() const /home/simon/code/mantid/Framework/Kernel/inc/MantidKernel/TimeSeriesProperty.h:79
    #1 0x7f24e2eda307 in Mantid::Kernel::TimeSeriesProperty<double>::nthValue(int) const /home/simon/code/mantid/Framework/Kernel/src/TimeSeriesProperty.cpp:1530
    #2 0x7f24e2e8b007 in Mantid::Kernel::filterByStatistic(Mantid::Kernel::TimeSeriesProperty<double> const*, Mantid::Kernel::Math::StatisticType) /home/simon/code/mantid/Framework/Kernel/src/TimeSeriesProperty.cpp:2183
    #3 0x7f24e3c1f0ab in Mantid::Geometry::XMLInstrumentParameter::createParamValue(Mantid::Kernel::TimeSeriesProperty<double>*) const /home/simon/code/mantid/Framework/Geometry/src/Instrument/XMLInstrumentParameter.cpp:139
    #4 0x7f24e6851c0a in operator double /home/simon/code/mantid/Framework/API/src/ExperimentInfo.cpp:224
    #5 0x7f24e685ac55 in Mantid::API::ExperimentInfo::populateWithParameter(Mantid::Geometry::ParameterMap&, std::string const&, Mantid::Geometry::XMLInstrumentParameter const&, Mantid::API::Run const&) /home/simon/code/mantid/Framework/API/src/ExperimentInfo.cpp:1219
    #6 0x7f24e68527f3 in Mantid::API::ExperimentInfo::populateInstrumentParameters() /home/simon/code/mantid/Framework/API/src/ExperimentInfo.cpp:303
    #7 0x7f24e69633e3 in Mantid::API::MatrixWorkspace::populateInstrumentParameters() /home/simon/code/mantid/Framework/API/src/MatrixWorkspace.cpp:842
    #8 0x7f24e592f814 in Mantid::DataHandling::LoadParameterFile::exec() /home/simon/code/mantid/Framework/DataHandling/src/LoadParameterFile.cpp:149
    #9 0x7f24e678af56 in Mantid::API::Algorithm::execute() /home/simon/code/mantid/Framework/API/src/Algorithm.cpp:563
    #10 0x8a25dd in apply_instrument_parameter_file_to_workspace(boost::shared_ptr<Mantid::API::MatrixWorkspace>, ScopedFileHelper::ScopedFile const&) /home/simon/code/mantid/Framework/Algorithms/test/NormaliseByDetectorTest.h:39
    #11 0x8b1dd5 in NormaliseByDetectorTestPerformance::setUp() /home/simon/code/mantid/Framework/Algorithms/test/NormaliseByDetectorTest.h:644
    #12 0xfa1d37 in CxxTest::RealTestDescription::setUp() /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/RealDescriptions.cpp:53
    #13 0xfa811a in CxxTest::TestRunner::runTest(CxxTest::TestDescription&) /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestRunner.h:91
    #14 0xfa7e03 in CxxTest::TestRunner::runSuite(CxxTest::SuiteDescription&) /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestRunner.h:79
    #15 0xfa79d3 in CxxTest::TestRunner::runWorld() /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestRunner.h:63
    #16 0xfa761e in CxxTest::TestRunner::runAllTests(CxxTest::TestListener&) /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestRunner.h:24
    #17 0xfb5357 in CxxTest::XUnitPrinter::run() /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/XUnitPrinter.h:30
    #18 0xfb6d8b in int CxxTest::Main<CxxTest::XUnitPrinter>(CxxTest::XUnitPrinter&, int, char**) /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestMain.h:106
    #19 0xfa0525 in main /home/simon/build/mantid-debug/Framework/Algorithms/test/AlgorithmsTest_runner.cpp:27
    #20 0x7f24e119ba3f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x20a3f)
    #21 0x4bb488 in _start (/home/simon/build/mantid-debug/bin/AlgorithmsTest+0x4bb488)

0x60e00011d5d8 is located 24 bytes to the right of 160-byte region [0x60e00011d520,0x60e00011d5c0)
allocated by thread T0 here:
    #0 0x7f24e8bbe1af in operator new(unsigned long) (/usr/lib/x86_64-linux-gnu/libasan.so.1+0x581af)
    #1 0x7f24e2cad7ce in __gnu_cxx::new_allocator<Mantid::Kernel::TimeValueUnit<double> >::allocate(unsigned long, void const*) /usr/include/c++/4.9/ext/new_allocator.h:104
    #2 0x7f24e2cace3a in std::allocator_traits<std::allocator<Mantid::Kernel::TimeValueUnit<double> > >::allocate(std::allocator<Mantid::Kernel::TimeValueUnit<double> >&, unsigned long) /usr/include/c++/4.9/bits/alloc_traits.h:357
    #3 0x7f24e2caba25 in std::_Vector_base<Mantid::Kernel::TimeValueUnit<double>, std::allocator<Mantid::Kernel::TimeValueUnit<double> > >::_M_allocate(unsigned long) /usr/include/c++/4.9/bits/stl_vector.h:170
    #4 0x7f24e2caa7ca in std::_Vector_base<Mantid::Kernel::TimeValueUnit<double>, std::allocator<Mantid::Kernel::TimeValueUnit<double> > >::_M_create_storage(unsigned long) /usr/include/c++/4.9/bits/stl_vector.h:185
    #5 0x7f24e2ca8a8a in std::_Vector_base<Mantid::Kernel::TimeValueUnit<double>, std::allocator<Mantid::Kernel::TimeValueUnit<double> > >::_Vector_base(unsigned long, std::allocator<Mantid::Kernel::TimeValueUnit<double> > const&) /usr/include/c++/4.9/bits/stl_vector.h:136
    #6 0x7f24e2ca6dd5 in std::vector<Mantid::Kernel::TimeValueUnit<double>, std::allocator<Mantid::Kernel::TimeValueUnit<double> > >::vector(std::vector<Mantid::Kernel::TimeValueUnit<double>, std::allocator<Mantid::Kernel::TimeValueUnit<double> > > const&) /usr/include/c++/4.9/bits/stl_vector.h:320
    #7 0x7f24e2ca58ad in Mantid::Kernel::TimeSeriesProperty<double>::TimeSeriesProperty(Mantid::Kernel::TimeSeriesProperty<double> const&) /home/simon/code/mantid/Framework/Kernel/inc/MantidKernel/TimeSeriesProperty.h:112
    #8 0x7f24e2ca3e8c in Mantid::Kernel::FilteredTimeSeriesProperty<double>::FilteredTimeSeriesProperty(Mantid::Kernel::TimeSeriesProperty<double>*, Mantid::Kernel::TimeSeriesProperty<bool> const&, bool) /home/simon/code/mantid/Framework/Kernel/src/FilteredTimeSeriesProperty.cpp:20
    #9 0x7f24e2dc01a8 in std::_MakeUniq<Mantid::Kernel::FilteredTimeSeriesProperty<double> >::__single_object std::make_unique<Mantid::Kernel::FilteredTimeSeriesProperty<double>, Mantid::Kernel::TimeSeriesProperty<double>*&, Mantid::Kernel::TimeSeriesProperty<bool> const&, bool const&>(Mantid::Kernel::TimeSeriesProperty<double>*&, Mantid::Kernel::TimeSeriesProperty<bool> const&, bool const&) /usr/include/c++/4.9/bits/unique_ptr.h:765
    #10 0x7f24e2dbaf44 in Mantid::Kernel::PropertyManager::filterByProperty(Mantid::Kernel::TimeSeriesProperty<bool> const&) /home/simon/code/mantid/Framework/Kernel/src/PropertyManager.cpp:167
    #11 0x7f24e6941c9a in Mantid::API::LogManager::filterByLog(Mantid::Kernel::TimeSeriesProperty<bool> const&) /home/simon/code/mantid/Framework/API/src/LogManager.cpp:230
    #12 0x7f24e55e5058 in Mantid::DataHandling::ISISRunLogs::addPeriodLogs(int, Mantid::API::Run&) /home/simon/code/mantid/Framework/DataHandling/src/ISISRunLogs.cpp:75
    #13 0x7f24e5811f7b in Mantid::DataHandling::LoadISISNexus2::createPeriodLogs(long, boost::shared_ptr<Mantid::DataObjects::Workspace2D>&) /home/simon/code/mantid/Framework/DataHandling/src/LoadISISNexus2.cpp:861
    #14 0x7f24e580ba71 in Mantid::DataHandling::LoadISISNexus2::exec() /home/simon/code/mantid/Framework/DataHandling/src/LoadISISNexus2.cpp:305
    #15 0x7f24e678af56 in Mantid::API::Algorithm::execute() /home/simon/code/mantid/Framework/API/src/Algorithm.cpp:563
    #16 0x7f24e5601f54 in Mantid::DataHandling::Load::loadSingleFile() /home/simon/code/mantid/Framework/DataHandling/src/Load.cpp:371
    #17 0x7f24e5601238 in Mantid::DataHandling::Load::exec() /home/simon/code/mantid/Framework/DataHandling/src/Load.cpp:337
    #18 0x7f24e678af56 in Mantid::API::Algorithm::execute() /home/simon/code/mantid/Framework/API/src/Algorithm.cpp:563
    #19 0x7f24e67de69c in Mantid::API::AlgorithmProxy::execute() /home/simon/code/mantid/Framework/API/src/AlgorithmProxy.cpp:71
    #20 0x8b0d42 in NormaliseByDetectorTestPerformance::setUp() /home/simon/code/mantid/Framework/Algorithms/test/NormaliseByDetectorTest.h:594
    #21 0xfa1d37 in CxxTest::RealTestDescription::setUp() /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/RealDescriptions.cpp:53
    #22 0xfa811a in CxxTest::TestRunner::runTest(CxxTest::TestDescription&) /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestRunner.h:91
    #23 0xfa7e03 in CxxTest::TestRunner::runSuite(CxxTest::SuiteDescription&) /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestRunner.h:79
    #24 0xfa79d3 in CxxTest::TestRunner::runWorld() /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestRunner.h:63
    #25 0xfa761e in CxxTest::TestRunner::runAllTests(CxxTest::TestListener&) /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestRunner.h:24
    #26 0xfb5357 in CxxTest::XUnitPrinter::run() /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/XUnitPrinter.h:30
    #27 0xfb6d8b in int CxxTest::Main<CxxTest::XUnitPrinter>(CxxTest::XUnitPrinter&, int, char**) /home/simon/code/mantid/Testing/Tools/cxxtest/cxxtest/TestMain.h:106
    #28 0xfa0525 in main /home/simon/build/mantid-debug/Framework/Algorithms/test/AlgorithmsTest_runner.cpp:27
    #29 0x7f24e119ba3f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x20a3f)

SUMMARY: AddressSanitizer: heap-buffer-overflow /home/simon/code/mantid/Framework/Kernel/inc/MantidKernel/TimeSeriesProperty.h:79 Mantid::Kernel::TimeValueUnit<double>::value() const
Shadow bytes around the buggy address:
  0x0c1c8001ba60: 00 00 00 00 fa fa fa fa fa fa fa fa 00 00 00 00
  0x0c1c8001ba70: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c1c8001ba80: fa fa fa fa fa fa fa fa 00 00 00 00 00 00 00 00
  0x0c1c8001ba90: 00 00 00 00 00 00 00 00 00 00 00 00 fa fa fa fa
  0x0c1c8001baa0: fa fa fa fa 00 00 00 00 00 00 00 00 00 00 00 00
=>0x0c1c8001bab0: 00 00 00 00 00 00 00 00 fa fa fa[fa]fa fa fa fa
  0x0c1c8001bac0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c1c8001bad0: 00 00 00 00 fa fa fa fa fa fa fa fa 00 00 00 00
  0x0c1c8001bae0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c1c8001baf0: fa fa fa fa fa fa fa fa 00 00 00 00 00 00 00 00
  0x0c1c8001bb00: 00 00 00 00 00 00 00 00 00 00 00 00 fa fa fa fa
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:       fa
  Heap right redzone:      fb
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack partial redzone:   f4
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Contiguous container OOB:fc
  ASan internal:           fe
```

### PR DESCRIPTION
**Pull request by dtasev**
## Description of work.

The NormaliseByDetector performance test now re-uses the code from the unit test, which does not cause a `heap-buffer-overflow`. However I have not specifically identified the root cause of the problem, so I am unsure if these changes could help solve the other `heap-buffer-overflow` issues.

**To test:**

Run `NormaliseByDetectorTest` compiled with -DCMAKE_BUILD_TYPE=Debug -DCMAKE_CXX_FLAGS=-fsanitize=address

Fixes #17443.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by DimitarTasev** [comment]
Restarted [Windows build 10527](http://builds.mantidproject.org/job/pull_requests-win7/10527/) due to unrelated test failure

File: Framework/Algorithms/test/NormaliseByDetectorTest.h [hunk 1]
```diff
@@ -155,37 +128,35 @@ class NormaliseByDetectorTest : public CxxTest::TestSuite {
...
+parameter file ontop of an existing instrument definition.
+The fit function is different for every detector.
+*/
+MatrixWorkspace_sptr create_workspace_with_detector_level_only_fit_functions() {
+  // Create a default workspace with no-fitting functions.
+  MatrixWorkspace_sptr ws = create_workspace_with_no_fitting_functions();
+  const std::string instrumentName = ws->getInstrument()->getName();
+
+  const double A1 = 1;
+  std::string componentLinks = "";
```
Comment (gvardany) [inline Framework/Algorithms/test/NormaliseByDetectorTest.h]: std::string componentLinks;

**Comment by DimitarTasev** [comment]
I don't think committing from GitHub started the build servers, will I have to push from a local change?

**Comment by DimitarTasev** [comment]
Restarted http://builds.mantidproject.org/job/pull_requests-win7/10705/ due to unrelated failure

**Comment by gvardany** [comment]
Could you fix the same at line 153?

**Review by gvardany — DISMISSED** [review]
The test passed without problem,

**Comment by martyngigg** [comment]
If I run the address sanitizer and pass it through `asan_symolize` then I get:

```
ilog=11, m_values.size()=10
=================================================================
==19191== ERROR: AddressSanitizer: heap-buffer-overflow on address 0x601c00115d18 at pc 0x7fc446f06a85 bp 0x7ffd554703d0 sp 0x7ffd554703c8
READ of size 8 at 0x601c00115d18 thread T0
    #0 0x7fc446f06a84 in value /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/Kernel/inc/MantidKernel/TimeSeriesProperty.h:79
    #1 0x7fc446edf51d in nthValue /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/Kernel/src/TimeSeriesProperty.cpp:1531 (discriminator 1)
    #2 0x7fc446e998a7 in filterByStatistic /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/Kernel/src/TimeSeriesProperty.cpp:2184
    #3 0x7fc447be7cb1 in createParamValue /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/Geometry/src/Instrument/XMLInstrumentParameter.cpp:139 (discriminator 1)
    #4 0x7fc44a72411c in operator double /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/API/src/ExperimentInfo.cpp:224
    #5 0x7fc44a72be2c in populateWithParameter /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/API/src/ExperimentInfo.cpp:1219
    #6 0x7fc44a724c72 in populateInstrumentParameters /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/API/src/ExperimentInfo.cpp:303
    #7 0x7fc44a810759 in populateInstrumentParameters /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/API/src/MatrixWorkspace.cpp:841
    #8 0x7fc449729249 in exec /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/DataHandling/src/LoadParameterFile.cpp:149
    #9 0x7fc44a670971 in execute /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/API/src/Algorithm.cpp:563
    #10 0x865a46 in _Z44apply_instrument_parameter_file_to_workspaceN5boost10shared_ptrIN6Mantid3API15MatrixWorkspaceEEERKN16ScopedFileHelper10ScopedFileE /mnt/data1/source/github/mantidproject/mantid/Framework/Algorithms/test/NormaliseByDetectorTest.h:39 (discriminator 1)
    #11 0x873c17 in setUp /mnt/data1/source/github/mantidproject/mantid/Framework/Algorithms/test/NormaliseByDetectorTest.h:644 (discriminator 1)
    #12 0xf0158d in setUp /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/RealDescriptions.cpp:53
    #13 0xf0774a in runTest /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestRunner.h:91
    #14 0xf074dd in runSuite /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestRunner.h:79
    #15 0xf07195 in runWorld /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestRunner.h:63
    #16 0xf06e5a in runAllTests /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestRunner.h:24
    #17 0xf13e3b in run /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/XUnitPrinter.h:30
    #18 0xf15483 in Main<CxxTest::XUnitPrinter> /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestMain.h:106
    #19 0xeffdb0 in main /mnt/data1/source/github/mantidproject/mantid/builds/debug/Framework/Algorithms/test/AlgorithmsTest_runner.cpp:27
    #20 0x7fc4453c4ec4 in __libc_start_main /build/buildd/eglibc-2.19/csu/libc-start.c:287
    #21 0x4b9318 in _start ??:?
0x601c00115d18 is located 24 bytes to the right of 160-byte region [0x601c00115c60,0x601c00115d00)
allocated by thread T0 here:
    #0 0x7fc44cab381a in _Znwm ??:?
    #1 0x7fc446ce8c18 in allocate /usr/include/c++/4.8/ext/new_allocator.h:104
    #2 0x7fc446ce7ba2 in _M_allocate /usr/include/c++/4.8/bits/stl_vector.h:168 (discriminator 1)
    #3 0x7fc446ce6afa in _M_create_storage /usr/include/c++/4.8/bits/stl_vector.h:181
    #4 0x7fc446ce4ef8 in _ZNSt12_Vector_baseIN6Mantid6Kernel13TimeValueUnitIdEESaIS3_EEC2EmRKS4_ /usr/include/c++/4.8/bits/stl_vector.h:136
    #5 0x7fc446ce348a in _ZNSt6vectorIN6Mantid6Kernel13TimeValueUnitIdEESaIS3_EEC2ERKS5_ /usr/include/c++/4.8/bits/stl_vector.h:312
    #6 0x7fc446ce1663 in _ZN6Mantid6Kernel18TimeSeriesPropertyIdEC2ERKS2_ /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/Kernel/inc/MantidKernel/TimeSeriesProperty.h:112
    #7 0x7fc446ce18bc in _ZN6Mantid6Kernel26FilteredTimeSeriesPropertyIdEC2EPNS0_18TimeSeriesPropertyIdEERKNS3_IbEEb /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/Kernel/src/FilteredTimeSeriesProperty.cpp:20
    #8 0x7fc446de91de in make_unique<Mantid::Kernel::FilteredTimeSeriesProperty<double>, Mantid::Kernel::TimeSeriesProperty<double>*&, const Mantid::Kernel::TimeSeriesProperty<bool>&, bool const&> /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/Kernel/inc/MantidKernel/make_unique.h:43
    #9 0x7fc446de4bb8 in filterByProperty /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/Kernel/src/PropertyManager.cpp:167
    #10 0x7fc44a7f0840 in filterByLog /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/API/src/LogManager.cpp:221
    #11 0x7fc449422e24 in addPeriodLogs /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/DataHandling/src/ISISRunLogs.cpp:75
    #12 0x7fc449622e1d in createPeriodLogs /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/DataHandling/src/LoadISISNexus2.cpp:861
    #13 0x7fc44961cdfd in exec /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/DataHandling/src/LoadISISNexus2.cpp:305 (discriminator 1)
    #14 0x7fc44a670971 in execute /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/API/src/Algorithm.cpp:563
    #15 0x7fc44943cfbe in loadSingleFile /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/DataHandling/src/Load.cpp:371
    #16 0x7fc44943c35d in exec /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/DataHandling/src/Load.cpp:337
    #17 0x7fc44a670971 in execute /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/API/src/Algorithm.cpp:563
    #18 0x7fc44a6bbe30 in execute /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Framework/API/src/AlgorithmProxy.cpp:71
    #19 0x872b7a in setUp /mnt/data1/source/github/mantidproject/mantid/Framework/Algorithms/test/NormaliseByDetectorTest.h:594 (discriminator 1)
    #20 0xf0158d in setUp /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/RealDescriptions.cpp:53
    #21 0xf0774a in runTest /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestRunner.h:91
    #22 0xf074dd in runSuite /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestRunner.h:79
    #23 0xf07195 in runWorld /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestRunner.h:63
    #24 0xf06e5a in runAllTests /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestRunner.h:24
    #25 0xf13e3b in run /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/XUnitPrinter.h:30
    #26 0xf15483 in Main<CxxTest::XUnitPrinter> /mnt/data1/source/github/mantidproject/mantid/builds/debug/../../Testing/Tools/cxxtest/cxxtest/TestMain.h:106
    #27 0xeffdb0 in main /mnt/data1/source/github/mantidproject/mantid/builds/debug/Framework/Algorithms/test/AlgorithmsTest_runner.cpp:27
    #28 0x7fc4453c4ec4 in __libc_start_main /build/buildd/eglibc-2.19/csu/libc-start.c:287
Shadow bytes around the buggy address:
  0x0c040001ab50: fa fa fa fa 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c040001ab60: 00 00 00 00 00 00 00 00 fa fa fa fa fa fa fa fa
  0x0c040001ab70: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c040001ab80: 00 00 00 00 fa fa fa fa fa fa fa fa 00 00 00 00
  0x0c040001ab90: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
=>0x0c040001aba0: fa fa fa[fa]fa fa fa fa 00 00 00 00 00 00 00 00
  0x0c040001abb0: 00 00 00 00 00 00 00 00 00 00 00 00 fa fa fa fa
  0x0c040001abc0: fa fa fa fa 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c040001abd0: 00 00 00 00 00 00 00 00 fa fa fa fa fa fa fa fa
  0x0c040001abe0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x0c040001abf0: 00 00 00 00 fa fa fa fa fa fa fa fa 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:     fa
  Heap righ redzone:     fb
  Freed Heap region:     fd
  Stack left redzone:    f1
  Stack mid redzone:     f2
  Stack right redzone:   f3
  Stack partial redzone: f4
  Stack after return:    f5
  Stack use after scope: f8
  Global redzone:        f9
  Global init order:     f6
  Poisoned by user:      f7
  ASan internal:         fe
==19191== ABORTING
```

This suggests a problem either in `TimeSeriesProperty::nthValue` or the calling code in `XMLInstrumentParameter::createParamValue`. It is not clear from the surface

**Comment by AndreiSavici** [comment]
Is more work being done on this PR to address @martyngigg 's comment?

**Comment by DimitarTasev** [comment]
@AndreiSavici It's not something I can work on right now, however I do plan to look into it more and finish it before 3.9 release

**Comment by NickDraper** [comment]
@DanNixon Can you pick this up as the developer

**Comment by gvardany** [comment]
Does anyone have time to revive this? If not we should close without merging. It's hanging for many releases now.

**Comment by martyngigg** [comment]
@gvardany You're right - I don't think anyone is available to look at this at the moment and there are now conflicts with more recent changes.

**Pull request closed without merging**